torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -0,0 +1,609 @@
+    mm_config: Float8MMConfig = Float8MMConfig(use_fast_accum=True)
+
+
+class Float8Tensor(TorchAOBaseTensor):
+    """
+    Float8 Quantized (weight) Tensor, with float8 dynamic quantization for activation or bfloat16 activation.
+
+    TODO: needs padding for cutlass kernels
+
+    Tensor Attributes:
+        _data: float8 raw data
+        scale: the scale for float8 Tensor
+
+    Non-Tensor Attributes:
+        block_size (List[int]): the block size for float8 quantization, meaning the shape of the elements
+        sharing the same set of quantization parameters (scale), have the same rank as _data or
+        is an empty list (representing per tensor quantization)
+        act_quant_kwargs (QuantizeTensorToFloat8Kwargs): the kwargs for Float8Tensor.to_float8
+        dtype: Original Tensor dtype
+    """
+
+    tensor_data_attrs = ["_data", "scale"]
+    tensor_attributes = ["block_size", "mm_config", "act_quant_kwargs", "dtype"]
+
+    def __new__(
+        cls,
+        _data,
+        scale,
+        block_size,
+        mm_config,
+        act_quant_kwargs,
+        dtype,
+    ):
+        shape = _data.shape
+        kwargs = {}
+        kwargs["device"] = _data.device
+        kwargs["dtype"] = dtype
+        kwargs["requires_grad"] = False
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)  # type: ignore[attr-defined]
+
+    def __init__(
+        self,
+        _data: torch.Tensor,
+        scale: torch.Tensor,
+        block_size: Optional[List[int]] = None,
+        mm_config: Float8MMConfig = Float8MMConfig(use_fast_accum=True),
+        act_quant_kwargs: Optional[QuantizeTensorToFloat8Kwargs] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        self._data = _data
+        self.scale = scale
+        self.block_size = block_size
+        self.mm_config = mm_config
+        self.act_quant_kwargs = act_quant_kwargs
+
+    def __tensor_flatten__(self):
+        return self.tensor_data_attrs, [
+            getattr(self, attr) for attr in self.tensor_attributes
+        ]
+
+    @classmethod
+    def __tensor_unflatten__(
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        tensors = [tensor_data_dict[name] for name in cls.tensor_data_attrs]
+        return cls(
+            *tensors,
+            *tensor_attributes,
+        )
+
+    def _apply_fn_to_data(self, fn):
+        tensors = [fn(getattr(self, attr)) for attr in self.tensor_data_attrs]
+        return self.__class__(
+            *tensors,
+            *[getattr(self, attr) for attr in self.tensor_attributes],
+        )
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}({self.act_quant_kwargs=}, {self._data=}, {self.scale=}, "
+            f"{self.block_size=}, {self.mm_config=}"
+            f"{self.shape=}, {self.device=}, {self.dtype=}, "
+            f"{self.requires_grad=})"
+        )
+
+    def _quantization_type(self):
```

Comment:
not sure where this is used, but should this include more information such as activation quantization info, etc?